### PR TITLE
feat: add turnover and cost sensitivity diagnostics

### LINKED ISSUE #38: Add turnover and cost sensitivity diagnostics
﻿## Summary
Make implementation cost part of the risk conversation by showing how results change under alternate turnover-cost assumptions without rerunning strategies.

## Scope
- add `evaluation.cost_sensitivity_bps: list[float] = []`
- add `cost_sensitivity.csv`
- recompute strategy summaries from existing gross-return and turnover paths under alternate bps assumptions
- add a compact reporting surface for cost sensitivity

## Dependencies
- part of Phase 5 epic #33
- depends on #36

## Acceptance
- cost sensitivity can be reviewed without changing the backtest engine contract
- the zero-cost case matches gross-return behavior
- alternate bps assumptions are visible per strategy in review artifacts

### PR DESCRIPTION
**Pull request by ricardogr07**
## Summary
- add cost sensitivity analytics driven from existing gross returns and turnover
- persist `cost_sensitivity.csv` for `backtest` and `run-experiment` and add a markdown report section
- cover the feature with unit and integration tests and document the new reporting surface

## Validation
- `python -m tox -e preflight` (timed out locally on this Windows machine)
- `python -m ruff check src tests README.md docs/architecture.md`
- `python -m mkdocs build --strict`
- `python -m build --no-isolation`
- `python -m pytest -q tests/unit/test_config.py tests/unit/test_analytics.py tests/unit/test_markdown_report.py --maxfail=5 --tb=short --basetemp C:\Users\ricar\AppData\Local\Temp\marketlab_pytest_pr38_unit2`
- `python -m pytest -q tests/integration/test_cost_sensitivity.py tests/integration/test_run_experiment.py::test_run_experiment_produces_baseline_and_ml_artifacts tests/integration/test_run_experiment.py::test_backtest_remains_baseline_only --maxfail=5 --tb=short --basetemp C:\Users\ricar\AppData\Local\Temp\marketlab_pytest_pr38_integration_focus2`

Closes #38

### CODE REVIEW

File: src/marketlab/reports/analytics.py [hunk 1]
```diff
@@ -215,6 +233,92 @@ def build_turnover_costs(performance: pd.DataFrame) -> pd.DataFrame:
...
+        )
+        .reset_index()
+        .sort_values("strategy")
+        .reset_index(drop=True)
+    )
+
+    scenario_frames: list[pd.DataFrame] = []
+    for bps_per_trade in _cost_sensitivity_grid(base_cost_bps, sensitivity_bps):
+        repriced = _repriced_performance(working, bps_per_trade)
+        metrics = compute_strategy_metrics(repriced)
```
[the remark this answers is not in the document]
Comment (ricardogr07) [reply src/marketlab/reports/analytics.py]: Handled in 22aa252. The cost-sensitivity path now computes repriced metrics with an explicit guard for non-positive final equity, so annualized return and sharpe-like become `NaN` instead of raising through complex annualization. I also added a regression test for a high-cost scenario that drives equity below zero.
